File: SDKs/gambitcomm.local/mimic/21.00/python/src/sdk/models/operations/protocol_syslog_get_statistics.py
from dataclasses import dataclass, field
from typing import List,Optional


@dataclass
class ProtocolSyslogGetStatisticsPathParams:
    agent_num: int = field(metadata={'path_param': { 'field_name': 'agentNum', 'style': 'simple', 'explode': False }})
    

@dataclass
class ProtocolSyslogGetStatisticsRequest:
    path_params: ProtocolSyslogGetStatisticsPathParams = field()
    

@dataclass
class ProtocolSyslogGetStatisticsResponse:
    content_type: str = field()
    status_code: int = field()
    protocol_syslog_get_statistics_200_application_json_int32_integers: Optional[List[int]] = field(default=None)
    

File: SDKs/gambitcomm.local/mimic/21.00/typescript/src/sdk/models/operations/protocolsysloggetstatistics.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class ProtocolSyslogGetStatisticsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=agentNum" })
  agentNum: number;
}


export class ProtocolSyslogGetStatisticsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ProtocolSyslogGetStatisticsPathParams;
}


export class ProtocolSyslogGetStatisticsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  protocolSyslogGetStatistics200ApplicationJsonInt32Integers?: number[];
}
